railties/lib/rails/test_unit/minitest_plugin.rb

```diff
@@ -1,14 +1,35 @@
-require "minitest"
 require "rails/test_unit/reporter"
 
-def Minitest.plugin_rails_init(options)
-  self.reporter << Rails::TestUnitReporter.new(options[:io], options)
-  if $rails_test_runner && (method = $rails_test_runner.find_method)
-    options[:filter] = method
+module Minitest
+  def self.plugin_rails_options(opts, options)
+    opts.separator "Usage: bin/rails test [options] [file or directory]"
+    opts.separator "You can run a single test by appending the line number to filename:"
+    opts.separator ""
+    opts.separator "bin/rails test test/models/user_test.rb:27"
+
+    opts.on("-e", "--environment [ENV]",
+            "Run tests in the ENV environment") do |env|
+      options[:environment] = env.strip
+    end
+
+    opts.on("-b", "--backtrace",
+            "Show the complete backtrace") do
+      options[:backtrace] = true
+    end
   end
 
-  if !($rails_test_runner && $rails_test_runner.show_backtrace?)
-    Minitest.backtrace_filter = Rails.backtrace_cleaner
+  def self.plugin_rails_init(options)
+    ENV["RAILS_ENV"] = options[:environment] || "test"
+
+    options[:patterns] = OptionParser.new(options[:args]).order!
+    Rails::TestRunner.require_files options[:patterns]
```

Comment:
I just realized we have this `require` call here, so do we need the `TestRunner.run` method at all?